ocaml/src/asd_server.ml

```diff
@@ -391,14 +392,44 @@ module Net_fd = struct
                   then todo
                   else buffer_size
                 in
-                Lwt_bytes.read fd_in        buffer 0 step >>= fun bytes_read ->
-                write_all socket buffer 0 bytes_read >>= fun () ->
+                Lwt_bytes.read fd_in buffer 0 step >>= fun bytes_read ->
+                write_all socket     buffer 0 bytes_read >>= fun () ->
                 loop (todo - bytes_read)
               end
           in
           loop size
        in
        Buffer_pool.with_buffer Buffer_pool.default_buffer_pool copy_using
+    | Rsocket socket ->
+       let copy_using buffer =
```

Comment:
You extracted a helper for this in https://github.com/openvstorage/alba/pull/181 which could be generalized a bit more to be usable here too